lib/bolt_server/transport_app.rb

```diff
@@ -675,6 +679,51 @@ def make_winrm_target(target_hash)
       end
     end
 
+    # Returns the base64 encoded tar archive of plugin code that is needed to calculate
+    # custom facts
+    #
+    # @param versioned_project [String] the versioned_project to build the plugin tarball from
+    post '/project_facts_plugin_tarball' do
+      raise BoltServer::RequestError, "'versioned_project' is a required argument" if params['versioned_project'].nil?
+      content_type :json
+      in_bolt_project(params['versioned_project']) do |context|
+        # Inspired by Bolt::Applicator.build_plugin_tarball
```

Comment:
`build_plugin_tarball` is basically "tar-up all of this stuff". That code will probably not change too much in the future and it seems basic enough that it's not worth introducing an implicit API requirement for Applicator to ensure that `build_plugin_tarball` isn't broken. So, I went with basically copy-pasting that code here.